### ISSUE TITLE
Use new GH action to compile and test for windows

### ISSUE THREAD

**Issue by wqferr** [lenscas/tealr#17]
I believe you mentioned in gitter you were having problems with the leafo Lua action. I recently bumped into the same issue and found [this one](https://github.com/marketplace/actions/setup-lua-luajit) which claims to work for all 3 major platforms.

Maybe worth having a look?

**Comment by lenscas** [comment]
OMG! THANKS!

That looks exactly like the thing I need :)

**Comment by wqferr** [comment]
Just had a look at the source code, currently it maps 5.4 to 5.3.5, so that's not reliable just yet.

Probably will be in the near future, but I'd say keep the current one for now until it fixes that.

**Comment by wqferr** [comment]
I created an issue on that, let's hope it's resolved soon.

**Comment by lenscas** [comment]
I only need a lua version that can run teal and luarocks to install it so having that mapping issue shouldn't be a problem.

However, I just looked and it seems like you need another action to make use of luarocks, and the action they link has this as the only open issue https://github.com/leafo/gh-actions-luarocks/issues/3 so that isn't very promising.

**Comment by wqferr** [comment]
I just realized we can use [hererocks](https://github.com/mpeterv/hererocks)! I've been using it to test stuff locally, didn't even think of using it in a GH action. It's a python module, so it's probably very easy to find a platform independent action.

**Comment by wqferr** [comment]
UPDATE: Hererocks works! I'm just having issues with LuaJIT on MacOS, but I'm pretty sure it's not relevant to your use case, so feel free to check how I did it in [my CI setup](https://github.com/wqferr/lua-vectorize/blob/51-compat/.github/workflows/main.yml). Note that this is not yet on master, so if you need to reference it later, be sure to follow this link or make sure you're in the `51-compat` branch.

**Comment by wqferr** [comment]
Did you work on this at all? If you have any questions feel free to tag me, my current CI setup is working smoothly.

**Comment by lenscas** [comment]
I made the changes in https://github.com/lenscas/tealr/pull/18 but it looks like windows is still not happy.

> message: Could not run `tl check`. Make sure it is available in the path: Os { code: 2, kind: NotFound, message: "The system cannot find the file specified." }

do I need to run something before `tl` becomes available in the PATH after installing it with luarocks on windows?

**Comment by wqferr** [comment]
TL;DR: Try running `. ./path_to_env/bin/activate` *before and after* running `luarocks install`.

Hererocks works like a virtualenv in python, so you need to run `. ./path_to_env/bin/activate` not only to export the new environment path, but also to update the shell on which executables are available. If you only ran it before running `luarocks install`, any new executables might not be recognized.

I haven't looked at your current setup, so you might have already done that, so sorry in advance as I'm kind of in a hurry right now.

**Comment by lenscas** [comment]
sadly, that did not fix it.

Weirdly enough, both linux and macos managed to pass so whatever it is, it is windows specific.

**Comment by wqferr** [comment]
Hm, maybe you could just use `.luaenv/bin/tl` instead of relying on PATH resolution?

**Comment by lenscas** [comment]
no, because `tl` gets executed by the code that needs to be tested.

**Comment by wqferr** [comment]
Hmm, maybe it's a behavior specific to cargo on windows? It might be overshadowing PATH? I can't think of anything else...
